src/components/Popover/index.tsx

```diff
@@ -117,17 +123,38 @@ export default function Popover({
   const updateCallback = useCallback(() => {
     update && update()
   }, [update])
-  useInterval(updateCallback, show ? 100 : null)
+  useInterval(updateCallback, show ? 200 : null)
+
+  useEffect(() => {
+    if (!delayBeforeShow) return
+    if (show) {
+      const tooltipTimer = setTimeout(() => {
+        setDelayShow(true)
+      }, delayBeforeShow)
+
+      return () => {
+        clearTimeout(tooltipTimer)
+      }
+    } else {
+      setDelayShow(false)
+      return
```

Comment:
nit: `return` is redundant here, isn't it?

Reply:
I get a lint error: Not all code paths return a value.ts(7030)